src/app/shared/components/search-bar/search-bar.component.ts

```diff
@@ -1,133 +1,69 @@
-          { label: 'Dilbeek', value: 'Dilbeek' },
-          { label: 'Heist-op-den-Berg', value: 'Heist-op-den-Berg' },
-          { label: 'Sint-Truiden', value: 'Sint-Truiden' },
-          { label: 'Lokeren', value: 'Lokeren' },
-          { label: "Braine-l'Alleud", value: "Braine-l'Alleud" },
-          { label: 'Geel', value: 'Geel' },
-          { label: 'Ninove', value: 'Ninove' },
-          { label: 'Grimbergen', value: 'Grimbergen' },
-          { label: 'Tongeren', value: 'Tongeren' },
-          { label: 'Zottegem', value: 'Zottegem' },
-          { label: 'Ieper', value: 'Ieper' },
-          { label: 'Halle', value: 'Halle' },
-          { label: 'Binche', value: 'Binche' },
-          { label: 'Lier', value: 'Lier' },
-          { label: 'Aarschot', value: 'Aarschot' },
-          { label: 'Morlanwelz', value: 'Morlanwelz' },
-          { label: 'Eupen', value: 'Eupen' },
-          { label: 'Huy', value: 'Huy' },
-          { label: 'Ath', value: 'Ath' },
-          { label: 'Neerpelt', value: 'Neerpelt' },
-          { label: 'Schoten', value: 'Schoten' },
-          { label: 'Menen', value: 'Menen' },
-          { label: 'Duffel', value: 'Duffel' },
-          { label: 'Dendermonde', value: 'Dendermonde' },
-          { label: 'Bastogne', value: 'Bastogne' },
-          { label: 'Malmedy', value: 'Malmedy' },
-          { label: 'Soignies', value: 'Soignies' },
-          { label: 'Knokke-Heist', value: 'Knokke-Heist' },
-          { label: 'Deinze', value: 'Deinze' },
-          { label: 'Wavre', value: 'Wavre' },
-          { label: 'Rixensart', value: 'Rixensart' },
-          { label: 'Oudenaarde', value: 'Oudenaarde' },
-          { label: 'Geraardsbergen', value: 'Geraardsbergen' },
-          { label: 'Hamme', value: 'Hamme' },
-          { label: 'Maaseik', value: 'Maaseik' },
-          { label: 'Châtelet', value: 'Châtelet' },
-          { label: 'Koksijde', value: 'Koksijde' },
-          { label: 'Tienen', value: 'Tienen' },
-          { label: 'Harelbeke', value: 'Harelbeke' },
-          { label: 'Bilzen', value: 'Bilzen' },
-          { label: 'Oupeye', value: 'Oupeye' },
-          { label: 'Jambes', value: 'Jambes' },
-          { label: 'Lommel', value: 'Lommel' },
-          { label: 'Westerlo', value: 'Westerlo' },
-          { label: 'Andenne', value: 'Andenne' },
-          { label: 'Flemalle', value: 'Flemalle' },
-          { label: 'Spa', value: 'Spa' },
-          { label: 'Lanaken', value: 'Lanaken' },
-        ],
-      },
-    ];
+    this.jobList?.subscribe((jobsList) => {
+      this.job = jobsList;
+      this.displayJob = jobsList;
+    });
   }
 
-  filterGroupedCity(event: AutoCompleteCompleteEvent) {
-    let query = event.query;
-    const filteredGroups = [];
-
-    if (!this.groupedCities) {
+  filterJobs() {
+    if (!this.searchCity && !this.searchJob) {
+      this.showResult = false;
       return;
     }
+    this.jobService.getAllJobs().subscribe((jobs) => {
```

Comment:
This whole logic needs to be reworked, you should have passed parameters inside getAllJobs() method